### ISSUE TITLE
Parameter sets broken in master?

### ISSUE THREAD

**Issue by tillkruss** [phpbench/phpbench#881]
When using c35e70b3a6ee7cdaf1a7048809b8e3dcb9f0f10f I'm seeing the following error:

```
\Relay\Benchmarks\MgetBench::benchMgetUsingPredis

    PhpBench\Executor\Exception\ExecutionError
    Fatal error: Uncaught TypeError: {closure}(): Argument #1 ($value) must be of type string, array given in /private/var/folders/cw/70xgjbbx7ps2mlhyqx24snbc0000gn/T/PhpBenchPz3t08:27
    Stack trace:
    #0 [internal function]: {closure}(Array)
    #1 /private/var/folders/cw/70xgjbbx7ps2mlhyqx24snbc0000gn/T/PhpBenchPz3t08(34): array_map(Object(Closure), Array)
    #2 {main}
      thrown in /private/var/folders/cw/70xgjbbx7ps2mlhyqx24snbc0000gn/T/PhpBenchPz3t08 on line 27


    #0 /Users/Till/Development/Repositories/relay/src/PhpBench/RedisExecutor.php(50): PhpBench\Executor\Benchmark\TemplateExecutor->execute(Object(PhpBench\Executor\ExecutionContext), Object(PhpBench\Registry\Config))
    #1 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/lib/Benchmark/Runner.php(366): Relay\PhpBench\RedisExecutor->execute(Object(PhpBench\Executor\ExecutionContext), Object(PhpBench\Registry\Config))
    #2 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/lib/Benchmark/Runner.php(308): PhpBench\Benchmark\Runner->runIteration(Object(Relay\PhpBench\RedisExecutor), Object(PhpBench\Registry\Config), Object(PhpBench\Model\Iteration), Object(PhpBench\Benchmark\Metadata\SubjectMetadata))
    #3 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/lib/Benchmark/Runner.php(279): PhpBench\Benchmark\Runner->runVariant(Object(Relay\PhpBench\RedisExecutor), Object(PhpBench\Registry\Config), Object(PhpBench\Benchmark\RunnerConfig), Object(PhpBench\Benchmark\Metadata\SubjectMetadata), Object(PhpBench\Model\Variant))
    #4 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/lib/Benchmark/Runner.php(197): PhpBench\Benchmark\Runner->runSubject(Object(Relay\PhpBench\RedisExecutor), Object(PhpBench\Benchmark\RunnerConfig), Object(PhpBench\Model\Subject), Object(PhpBench\Benchmark\Metadata\SubjectMetadata))
    #5 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/lib/Benchmark/Runner.php(123): PhpBench\Benchmark\Runner->runBenchmark(Object(PhpBench\Benchmark\RunnerConfig), Object(PhpBench\Model\Benchmark), Object(PhpBench\Benchmark\Metadata\BenchmarkMetadata))
    #6 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/lib/Console/Command/Handler/RunnerHandler.php(141): PhpBench\Benchmark\Runner->run(Object(Generator), Object(PhpBench\Benchmark\RunnerConfig))
    #7 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/lib/Console/Command/RunCommand.php(134): PhpBench\Console\Command\Handler\RunnerHandler->runFromInput(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\StreamOutput), Object(PhpBench\Benchmark\RunnerConfig))
    #8 /Users/Till/Development/Repositories/relay/vendor/symfony/console/Command/Command.php(299): PhpBench\Console\Command\RunCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\StreamOutput))
    #9 /Users/Till/Development/Repositories/relay/vendor/symfony/console/Application.php(978): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\StreamOutput))
    #10 /Users/Till/Development/Repositories/relay/vendor/symfony/console/Application.php(295): Symfony\Component\Console\Application->doRunCommand(Object(PhpBench\Console\Command\RunCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\StreamOutput))
    #11 /Users/Till/Development/Repositories/relay/vendor/symfony/console/Application.php(167): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\StreamOutput))
    #12 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/lib/PhpBench.php(51): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\StreamOutput))
    #13 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/bin/phpbench.php(25): PhpBench\PhpBench::run()
    #14 /Users/Till/Development/Repositories/relay/vendor/phpbench/phpbench/bin/phpbench(4): require('/Users/Till/Dev...')
    #15 {main}
```

This is how the `@ParamProviders` used to work:

```php
class GetThroughputBench extends BenchCase
{
    /**
     * @ParamProviders("provideKeys")
     */
    public function benchGetThroughputOfRelayWarmed($params)
    {
        foreach ($params['keys'] as $key) {
            $this->relay->get($key);
        }
    }

    /**
     * Provides the keys for each benchmark.
     *
     * @return Generator
     */
    public function provideKeys()
    {
        $keys = [];

        foreach (static::loadJson('dataset-meteorites.json') as $landing) {
            $keys[] = "meteorite:{$landing->id}";
        }

        yield 'dataset' => [
            'keys' => $keys,
        ];
    }
}
```

**Comment by dantleech** [comment]
yes - or at least if you're using a custom executor. I created a task for this in the 1.1 project, but will fix against this ticket.

**Comment by dantleech** [comment]
Is your benchmark suite on public and on github?

**Comment by tillkruss** [comment]
@dantleech It is: https://github.com/cachewerk/relay

**Comment by tillkruss** [comment]
@dantleech Is there anything I can do easily to fix it on my end?

**Comment by dantleech** [comment]
At the moment using the 1.0.x branch (i.e. stable) I will make the parameter changes B/C in the master branch

**Comment by dantleech** [comment]
I don't get an error on `master`:

```
PHPBench (dev-master) running benchmarks...
with configuration file: /home/daniel/www/cachewerk/relay/phpbench.json
with PHP version 7.4.14, xdebug ✔, opcache ❌

\Relay\Benchmarks\MgetBench

    benchMgetUsingPredis....................I9 - Mo0.003ms (±21.43%)
    benchMgetUsingCredis....................I9 - Mo0.003ms (±39.90%)
    benchMgetUsingPhpRedis..................I9 - Mo0.003ms (±21.08%)
    benchMgetUsingRelay.....................I9 - Mo0.004ms (±26.80%)
    benchMgetUsingRelayWarmed...............I9 - Mo0.001ms (±41.46%)

Subjects: 5, Assertions: 0, Failures: 0, Errors: 0
```

(i stripped the deps to run without any extensions/redis etc, but the param providers are the same)

**Comment by tillkruss** [comment]
Sorry, `phpbench/phpbench` is locked at 4ecdb15.

```
git clone git@github.com:cachewerk/relay.git relay-test
cd relay-test
composer install
...
- Installing phpbench/phpbench (dev-master 4ecdb15):
```

If you strip the commit from:
```json
{ "require-dev": { "phpbench/phpbench": "dev-master#4ecdb15" }}
```

The it errors.

**Comment by dantleech** [comment]
Confirmed, thanks

**Comment by dantleech** [comment]
Fixed in #886, but need still to update the API to maintain compatiblity with 1.0 but don't think that affects this issue.

**Comment by tillkruss** [comment]
That resolved it. Thank you @dantleech!
